messaging/tck/src/main/java/org/eclipse/microprofile/reactive/messaging/tck/arquillian/TckArquillianDeployListener.java

```diff
@@ -0,0 +1,141 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package org.eclipse.microprofile.reactive.messaging.tck.arquillian;
+
+import org.eclipse.microprofile.reactive.messaging.tck.ContainerController;
+import org.eclipse.microprofile.reactive.messaging.tck.MockedReceiver;
+import org.eclipse.microprofile.reactive.messaging.tck.MockedSender;
+import org.eclipse.microprofile.reactive.messaging.tck.ReceiveTimeoutException;
+import org.eclipse.microprofile.reactive.messaging.tck.TckMessagingManager;
+import org.eclipse.microprofile.reactive.messaging.tck.spi.TestEnvironment;
+import org.eclipse.microprofile.reactive.messaging.tck.mocks.MockPayload;
+import org.eclipse.microprofile.reactive.messaging.tck.mocks.SimpleMessage;
+import org.eclipse.microprofile.reactive.messaging.tck.spi.TckContainer;
+import org.jboss.arquillian.container.spi.client.container.DeployableContainer;
+import org.jboss.arquillian.container.spi.client.container.DeploymentException;
+import org.jboss.arquillian.container.spi.client.deployment.DeploymentDescription;
+import org.jboss.arquillian.container.spi.event.container.AfterUnDeploy;
+import org.jboss.arquillian.container.spi.event.container.BeforeDeploy;
+import org.jboss.arquillian.core.api.annotation.Observes;
+import org.jboss.arquillian.test.spi.TestClass;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.ServiceLoader;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class TckArquillianDeployListener {
+
+  private final TckContainer tckContainer;
+
+  private final Map<Class<?>, List<DeploymentDescription>> activeDeployments = new ConcurrentHashMap<>();
+
+  /**
+   * I don't know how deployments work in arquillian, I assume I need to explicitly deploy everything?
+   */
+  private final Archive<?> frameworkArchive = ShrinkWrap.create(JavaArchive.class)
+      .addClass(MockedReceiver.class)
```

Comment:
you could do `addPackage(MockedReceiver.class.getPackage())` that adds all from the pgk

Reply:
Thanks, will do.